sr_hand_health_report/src/sr_hand_health_report/backlash_check.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+from urdf_parser_py.urdf import URDF
+
+
+class BacklashCheck(SrHealthReportCheck):
+
+    TEST_COUNT = 10
+    THRESHOLD = 0.01
+    FINGER_PWM = -250
+    WIGGLING_TIME = 2
+    FINGERS = ("TH", "FF", "MF", "RF", "LF")
+    PASSED_THRESHOLDS = {'std': 0.015, 'avg': 0.01}
+
+    def __init__(self, hand_side, fingers_to_test):
+        """
+            Initialize the BacklashCheck object
+            @param hand_side: String indicating the side
+            @param fingers_to_test: List of finger prefixes to test
+        """
+        super().__init__(hand_side, fingers_to_test)
+        self._side_sign_map = {"ff": -1, "mf": -1, "rf": 1, "lf": 1, "th": 1, "wr": 1}
+        self._name = "backlash"
+        self._result = {'backlash': {}}
+        self.joint_limits = {}
+        self._set_joint_limits()
+
+    def _set_joint_limits(self):
+        """
+            Sets the joint limits from the URDF
+        """
+        joint_list = [joint_name.lower() for joint_name in self._joint_msg.name]
+        for joint in URDF.from_parameter_server().joints:
+            if joint.name.lower() in joint_list:
+                self.joint_limits[joint.name.lower()] = joint.limit
+
+    def move_fingers_to_start_position(self):
+        """
+            Moves tested fingers into 0-degree joint anglees in position control mode.
+        """
+        self.switch_controller_mode('position')
+        for finger in self.fingers_to_check:
+            finger.move_finger(0, 'position')
+            rospy.logwarn(f"Moving to start {finger.finger_name}")
+
+    def run_check(self):
+        """
+            Runs the backlash check for all selected fingers and assings the result into the _result variable.
+        """
+        self._result = {'backlash': {}}
+        if self._stopped_execution:
+            self._stopped_execution = False
+            return
+        rospy.loginfo("Running Backlash Check")
+        result = dict(self._result)
+
+        self.move_fingers_to_start_position()
+
+        self.switch_controller_mode("position")
+        for finger_object in self.fingers_to_check:
+            self.move_finger_to_side(finger_object, 'right')
+
+        for finger_object in self.fingers_to_check:
+            finger_object.move_finger(0, 'position')
+            self.switch_controller_mode("effort")
+
+            for joint in finger_object.joints_dict.values():
+                rospy.logwarn(f"Running check for {joint.joint_name}")
+                if finger_object.finger_name in ('ff', 'mf', 'rf', 'lf') and joint.joint_index != "j1":
+                    joint_result = self.wiggle_joint(joint)
+                    result['backlash'][joint.joint_name] = joint_result
+                elif finger_object.finger_name in ('th', 'wr'):
+                    joint_result = self.wiggle_joint(joint)
+                    result['backlash'][joint.joint_name] = joint_result
+
+                if self._stopped_execution:
+                    break
```

Comment:
would changing this break for a return and remove the next if self._stopped_execution: return make any difference in the behaviour?

Reply:
so if I return, I won't execute the last two in this method, which is setting the _result and _stopped_execution variables necessary to interact with the GUI correctly. The scope of 'break' is only to the most 'inner' for/while loop, so breaking the joint loop will still iterate over the fingers (moving fingers and switching controllers in this case - which I don't want to). I know it's probably not the best solution but couldn't find a better solution. Happy to change though if you have a suggestion!